src/Tizen.NUI/src/internal/WebView/WebHttpRequestInterceptor.cs

```diff
@@ -46,56 +51,143 @@ public string Url
+        /// When application doesn't have a response for intercepted request,
+        /// this function would be called which notifies engine to proceed with normal resource loading.
+        /// It can be called only INSIDE WebContext.HttpRequestIntercepted.
+        /// After it called, any further call on WebHttpRequestInterceptor results in undefined behavior.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public bool Ignore()
         {
-            bool result = Interop.WebHttpRequestInterceptor.Ignore(SwigCPtr);
+            bool result = Interop.WebHttpRequestInterceptor.Ignore(interceptorHandle);
             if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
             return result;
         }
 
         /// <summary>
         /// Sets status code and status text of response for intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
         /// <param name="statusCode">Status code of response</param>
         /// <param name="customStatusText">Status text of response</param>
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public bool SetResponseStatus(int statusCode, string customStatusText)
         {
-            bool result = Interop.WebHttpRequestInterceptor.SetResponseStatus(SwigCPtr, statusCode, customStatusText);
+            bool result = Interop.WebHttpRequestInterceptor.SetResponseStatus(interceptorHandle, statusCode, customStatusText);
             if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
             return result;
         }
 
         /// <summary>
         /// Adds HTTP header to response for intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
         /// <param name="fieldName">Key of response header</param>
         /// <param name="fieldValue">Value of response header</param>
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public bool AddResponseHeader(string fieldName, string fieldValue)
         {
-            bool result = Interop.WebHttpRequestInterceptor.AddResponseHeader(SwigCPtr, fieldName, fieldValue);
+            bool result = Interop.WebHttpRequestInterceptor.AddResponseHeader(interceptorHandle, fieldName, fieldValue);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return result;
+        }
+
+        /// <summary>
+        /// Adds HTTP headers to response for intercepted request.
+        /// This function can be used inside or outside WebContext.HttpRequestIntercepted.
+        /// <param name="headers">Map of response headers</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool AddResponseHeaders(PropertyMap headers)
```

Comment:
How about use IDictionary<string,string>?
because, Headers will be comes from WebHttpRequestInterceptor.Headers API and some values need to update to use on `AddResponseHeaders` but PropertyMap does not support update value
```suggestion
        public bool AddResponseHeaders(IDictionary<string,string> header)
```

Reply:
If need to add a special cookie value 

Dictionary version
``` c#
HttpRequestIntercepted += (s, e) => {
    var header = e.HttpRequestInterceptor.Header;
    if (header.ContainKey("Cookie")) {
          header["Cookie"] += ";mycookie=myvalue";
    } else {
          header["Cookie"] += "mycookie=myvalue";
    }
    AddResponseHeaders(header);
};
```

PropertyMap version
``` c#
HttpRequestIntercepted += (s, e) => {
    var header = e.HttpRequestInterceptor.Header;
    PropertyMap newHeader = new PropertyMap();
    bool hasCookie = false;
    for (int i =0; i < header.Count(); i++) {
         var key = header.GetKeyAt(i);
         if (key.StringKey == "Cookie") {
              hasCookie = true;
              header.GetValue(i).Get(out string cookieValues);
              cookieValues += ";mycookie=myvalue";
         } else {
            newHeader.Insert(key.StringKey, header.GetValue(i));
         }
    }
    if (!hasCookie) {
         newHeader.Insert("Cookie", new PropertyValue("mycookie=myvalue"));
    }
    AddResponseHeaders(newHeader);
};
```

Reply:
Done.

Reply:
If you don't mind, could you change `Dictionary` to `IDictionary` as he originally suggested?

Reply:
Done.